src/lifecycle.rs

```diff
@@ -73,4 +91,12 @@ impl InstanceData {
     pub(crate) fn drop_queue<'a, C: Context<'a>>(cx: &mut C) -> Arc<ThreadsafeFunction<NapiRef>> {
         Arc::clone(&InstanceData::get(cx).drop_queue)
     }
+
+    /// Helper to return a reference to the `invoke_callback` field of `InstanceData`
+    #[cfg(feature = "event-queue-api")]
+    pub(crate) fn threadsafe_trampoline<'a, C: Context<'a>>(
+        cx: &mut C,
+    ) -> Arc<RwLock<ThreadsafeTrampoline>> {
+        Arc::clone(&InstanceData::get(cx).threadsafe_trampoline)
```

Comment:
Nitpick: it might be nice to lazily initialize this for programs that don't use EventQueues.

Reply:
It sounds like in order to that I'll have to store it in an `Arc<RwLock<Option<Arc<RwLock<ThreadsafeTrampoline>>>>>` which doesn't look to nice to me. Is there any better way to do this?

Reply:
You have an exclusive (`&mut`) reference to InstanceData, so you don't need extra locking. `Option<Arc<_>>` is sufficient.